internal/runtime/catalog/catalog.go

```diff
@@ -0,0 +1,333 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package catalog
+
+import (
+	"fmt"
+	"reflect"
+	goruntime "runtime"
+	"strings"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/naming"
+	"k8s.io/kube-openapi/pkg/common"
+)
+
+// Catalog contains all information about RuntimeHooks defined in Cluster API,
+// including mappings between RuntimeHook functions and the corresponding GroupVersionHooks,
+// metadata and OpenAPIDefinitions.
+type Catalog struct {
+	// scheme allows access to api-machinery utilities to implement conversions of
+	// request and response types.
+	scheme *runtime.Scheme
+
+	// gvhToType allows mapping a GroupVersionHook to the corresponding RuntimeHook function.
```

Comment:
```suggestion
	// gvhToType maps a GroupVersionHook to the corresponding RuntimeHook function.
```
Similar below with comments including 'allows'